clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -422,12 +422,108 @@ bool Sema::isKnownGoodSYCLDecl(const Decl *D) {
   return false;
 }
 
-static bool isZeroSizedArray(QualType Ty) {
-  if (const auto *CATy = dyn_cast<ConstantArrayType>(Ty))
-    return CATy->getSize() == 0;
+static bool isZeroSizedArray(Sema &SemaRef, QualType Ty) {
```

Comment:
Can you please explain why this change was required?

Reply:
I don't quite remember all the details, so my explanation may not be quite correct, but what I remember is that we shouldn't do `dyn_cast` on types because in AST just say type A and typedef-ed/qualified/sugared type A are not same types. The right way to do it like we did in the past - https://github.com/intel/llvm/pull/2082/files .

Reply:
Oh yea. I remember now - https://github.com/intel/llvm/pull/2072#discussion_r452201923